addons/pos_self_order/controllers/self_entry.py

```diff
@@ -6,8 +6,16 @@
 
 
 class PosSelfKiosk(http.Controller):
-    @http.route(["/pos-self/<config_id>", "/pos-self/<config_id>/<path:subpath>"], auth="public", website=True, sitemap=True)
-    def start_self_ordering(self, config_id=None, access_token=None, table_identifier=None):
+    @http.route(["/pos-self/<config_id>", "/pos-self/<config_id>/<path:subpath>" ], auth="public", website=True, sitemap=True)
```

Comment:
given that the access_token and the table identifier are optional parameters, why would we not use optional parameters?

If the goal is to make the url shorter we can just replace `access_token` with `a` and `table_identifier` with `t`, i.e. `mydb.odoo.com/pos-self?a=123&t=1`

Reply:
doing this means that no change to the router will be needed

Reply:
need to be able to handle url

/pos-self/config_id/access_token/table_identifier/others

and

/pos_self/config_id/others?access_token=blabla&table_identifier=blabla

Reply:
you can simply have 
```py
def start_self_ordering(self, config_id=None, subpath=None, access_token=None, table_identifier=None, a=None, t=None):
    access_token = access_token or a
    table_identifier = table_identifier or t
```

Reply:
see https://www.odoo.com/odoo/my-tasks/4106628

Reply:
is to match the new url scheme of odoo

Reply:
then the base64 option should work well for this